src/Altcha.php

```diff
@@ -189,63 +172,50 @@ private static function extractParams(Payload $payload): array
         return $computedHash === $fieldsHash;
     }
 
 
     /**
      * Verifies the server signature.
      *
-     * @param string|array<array-key, mixed> $data    The payload to verify (string or `ServerSignaturePayload` array).
-     * @param string                         $hmacKey The HMAC key used for verification.
+     * @param string|array<array-key, mixed> $data The payload to verify (string or `ServerSignaturePayload` array).
      */
-    public static function verifyServerSignature($data, string $hmacKey): ServerSignatureVerification
+    public function verifyServerSignature(string|array $data): ServerSignatureVerification
     {
-        $payload = self::verifyAndBuildServerSignaturePayload($data);
+        $payload = $this->verifyAndBuildServerSignaturePayload($data);
 
         if (!$payload) {
             return new ServerSignatureVerification(false, null);
         }
 
-        $hash = self::hash($payload->algorithm, $payload->verificationData);
-        $expectedSignature = self::hmacHex($payload->algorithm, $hash, $hmacKey);
+        $hash = $this->hasher->hash($payload->algorithm, $payload->verificationData);
+        $expectedSignature = $this->hasher->hashHmacHex($payload->algorithm, $hash, $this->hmacKey);
 
         parse_str($payload->verificationData, $params);
 
-        $classification = isset($params['classification']) && is_string($params['classification']) ? $params['classification'] : '';
-        $country = isset($params['country']) && is_string($params['country']) ? $params['country'] : '';
-        $detectedLanguage = isset($params['detectedLanguage']) && is_string($params['detectedLanguage']) ? $params['detectedLanguage'] : '';
-        $email = isset($params['email']) && is_string($params['email']) ? $params['email'] : '';
-        $expire = isset($params['expire']) && is_numeric($params['expire']) ? (int) $params['expire'] : 0;
-        $fields = isset($params['fields']) && is_array($params['fields']) ? $params['fields'] : [];
-        $fieldsHash = isset($params['fieldsHash']) && is_string($params['fieldsHash']) ? $params['fieldsHash'] : '';
-        $reasons = isset($params['reasons']) && is_array($params['reasons']) ? $params['reasons'] : [];
-        $score = isset($params['score']) && is_numeric($params['score']) ? (float) $params['score'] : 0.0;
-        $time = isset($params['time']) && is_numeric($params['time']) ? (int) $params['time'] : 0;
-        $verified = isset($params['verified']) && $params['verified'];
-
         $verificationData = new ServerSignatureVerificationData(
-            $classification,
-            $country,
-            $detectedLanguage,
-            $email,
-            $expire,
-            $fields,
-            $fieldsHash,
-            $reasons,
-            $score,
-            $time,
-            $verified,
+            classification: isset($params['classification']) && is_string($params['classification']) ? $params['classification'] : '',
```

Comment:
This is just to make the code easier to read. Still not optimal, but close to the original.